tests/next_tests/integration_tests/feature_tests/dace/orchestration.py

```diff
@@ -0,0 +1,100 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import numpy as np
+from typing import Optional
+from types import ModuleType
+import pytest
+
+import gt4py.next as gtx
+from gt4py.next import backend as next_backend
+
+from next_tests.integration_tests import cases
+from next_tests.integration_tests.cases import IDim, Ioff, JDim, Joff, cartesian_case
+from next_tests.integration_tests.feature_tests.ffront_tests.ffront_test_utils import (
+    exec_alloc_descriptor,
+)
+
+try:
+    import dace
+    from gt4py.next.program_processors.runners.dace import run_dace_cpu, run_dace_gpu
+except ImportError:
+    dace: Optional[ModuleType] = None  # type:ignore[no-redef]
+    run_dace_cpu: Optional[next_backend.Backend] = None
+    run_dace_gpu: Optional[next_backend.Backend] = None
+
+
+@gtx.field_operator
+def lap(in_field: gtx.Field[[IDim, JDim], "float"]) -> gtx.Field[[IDim, JDim], "float"]:
+    return (
+        -4.0 * in_field
+        + in_field(Ioff[1])
+        + in_field(Joff[1])
+        + in_field(Ioff[-1])
+        + in_field(Joff[-1])
+    )
+
+
+@gtx.field_operator
+def laplap(in_field: gtx.Field[[IDim, JDim], "float"]) -> gtx.Field[[IDim, JDim], "float"]:
+    return lap(lap(in_field))
+
+
+@gtx.program
+def lap_program(
+    in_field: gtx.Field[[IDim, JDim], "float"], out_field: gtx.Field[[IDim, JDim], "float"]
+):
+    lap(in_field, out=out_field[1:-1, 1:-1])
+
+
+@gtx.program
+def laplap_program(
+    in_field: gtx.Field[[IDim, JDim], "float"], out_field: gtx.Field[[IDim, JDim], "float"]
+):
+    laplap(in_field, out=out_field[2:-2, 2:-2])
+
+
+def lap_ref(inp):
+    """Compute the laplacian using numpy"""
+    return -4.0 * inp[1:-1, 1:-1] + inp[:-2, 1:-1] + inp[2:, 1:-1] + inp[1:-1, :-2] + inp[1:-1, 2:]
+
+
+def test_sdfgConvertible_laplap(cartesian_case):
```

Comment:
Consider adding a test calling the same program passing different offset providers

Reply:
Could you please elaborate a bit more. I am not sure I get it. Thanks.

Reply:
I meant that the current test is just a single programs without arguments which is called only once, so problems related to capturing too much state in the program closure at compile time cannot be detected.

I would change or extent the test program to receive all the arguments from outside (fields and offset_provider) and then call the program with completely different arguments to effectively test that the closure is not capturing run-time information.

Reply:
I have updated the test. Thanks for the feedback!

Keep in mind that this PR implements the SDFGConvertible interface for the GT4Py programs. Therefore, the test just checks that a program placed inside an DaCe Program can be parsed as a nested SDFG.

The closure features and the runtime behavior will be properly taken into account in the Orchestration PR in icon4py.